### PR TITLE
Cherrypick: support multiple calls per comment

### LINKED ISSUE #21714: cherrypick plugin doesn't response to all commands in a github comment
**What happened**:
cherry-pick plugin didn't detect multiple uses of `/cherry-pick` command in the same message

ie. from https://github.com/knative/serving/pull/11127#issuecomment-814917770
```
/cherry-pick release-0.22
/cherry-pick release-0.21
```
**What you expected to happen**:
It would schedule two cherry picks

**How to reproduce it (as minimally and precisely as possible)**:

**Please provide links to example occurrences, if any**:

**Anything else we need to know?**:

### PR DESCRIPTION
**Pull request by matthyx**
fix #21714

### CODE REVIEW

File: prow/external-plugins/cherrypicker/server.go [hunk 1]
```diff
@@ -289,12 +289,13 @@ func (s *Server) handlePullRequest(l *logrus.Entry, pre github.PullRequestEvent)
 		if len(cherryPickMatches) == 0 || len(cherryPickMatches[0]) != 2 {
```
Comment (alvaroaleman) [inline prow/external-plugins/cherrypicker/server.go:289]: Don't we need to move the second check into the loop?

Comment (matthyx) [reply prow/external-plugins/cherrypicker/server.go:289]: I think we can remove it completely... either it matches and match[1] exists, or it doesn't and we won't iterate.

Comment (matthyx) [reply prow/external-plugins/cherrypicker/server.go:289]: I mean this would be silly, right?
```
			if len(match) != 2 {
				continue
			}
```

Comment (alvaroaleman) [reply prow/external-plugins/cherrypicker/server.go:289]: I guess you are right, if it matches the capture group will have a value so we can just remove the check

**Review by alvaroaleman — APPROVED** [review]

**Pull request merged**